gcloud-java-storage/src/main/java/com/google/gcloud/storage/Bucket.java

```diff
@@ -125,6 +129,287 @@ public static BucketSourceOption metagenerationNotMatch() {
+        case PREDEFINED_ACL:
+          return StorageRpc.Tuple.of(blobInfo,
+              Storage.BlobTargetOption.predefinedAcl((Storage.PredefinedAcl) value()));
+        case IF_GENERATION_MATCH:
+          blobId = BlobId.of(blobId.bucket(), blobId.name(), (Long) value());
+          return StorageRpc.Tuple.of(blobInfo.toBuilder().blobId(blobId).build(),
+              Storage.BlobTargetOption.generationMatch());
+        case IF_GENERATION_NOT_MATCH:
+          blobId = BlobId.of(blobId.bucket(), blobId.name(), (Long) value());
+          return StorageRpc.Tuple.of(blobInfo.toBuilder().blobId(blobId).build(),
+              Storage.BlobTargetOption.generationNotMatch());
+        case IF_METAGENERATION_MATCH:
+          return StorageRpc.Tuple.of(blobInfo.toBuilder().metageneration((Long) value()).build(),
+              Storage.BlobTargetOption.metagenerationMatch());
+        case IF_METAGENERATION_NOT_MATCH:
+          return StorageRpc.Tuple.of(blobInfo.toBuilder().metageneration((Long) value()).build(),
+              Storage.BlobTargetOption.metagenerationNotMatch());
+        default:
+          throw new AssertionError("Unexpected enum value");
+      }
+    }
+
+    /**
+     * Returns an option for specifying blob's predefined ACL configuration.
+     */
+    public static BlobTargetOption predefinedAcl(Storage.PredefinedAcl acl) {
+      return new BlobTargetOption(StorageRpc.Option.PREDEFINED_ACL, acl);
+    }
+
+    /**
+     * Returns an option that causes an operation to succeed only if the target blob does not exist.
+     * This option can not be provided together with {@link #generationMatch(long)} or
+     * {@link #generationNotMatch(long)}.
+     */
+    public static BlobTargetOption doesNotExist() {
+      return new BlobTargetOption(StorageRpc.Option.IF_GENERATION_MATCH, 0L);
+    }
+
+    /**
+     * Returns an option for blob's data generation match. If this option is used the request will
+     * fail if generation does not match the provided value. This option can not be provided
+     * together with {@link #generationNotMatch(long)} or {@link #doesNotExist()}.
+     */
+    public static BlobTargetOption generationMatch(long generation) {
+      return new BlobTargetOption(StorageRpc.Option.IF_GENERATION_MATCH, generation);
+    }
+
+    /**
+     * Returns an option for blob's data generation mismatch. If this option is used the request
+     * will fail if blob's generation matches the provided value. This option can not be provided
+     * together with {@link #generationMatch(long)} or {@link #doesNotExist()}.
+     */
+    public static BlobTargetOption generationNotMatch(long generation) {
+      return new BlobTargetOption(StorageRpc.Option.IF_GENERATION_NOT_MATCH, generation);
+    }
+
+    /**
+     * Returns an option for blob's metageneration match. If this option is used the request will
+     * fail if metageneration does not match the provided value. Either this option or
```

Comment:
"Either this ..." Why did you choose a different phrasing than what you did before "This option can not be ..."?
I would keep phrasing the same and I think the "This option can not .." is more clear.